From 11751645df1fcb6dc59a9e095d15894900a795b3 Mon Sep 17 00:00:00 2001
From: U2FsdGVkX1 <U2FsdGVkX1@gmail.com>
Date: Fri, 26 Jul 2024 19:22:08 +0800
Subject: [PATCH] adjust label style

---
 src/core/lv_refr.c | 32 ++++++++++++++++++++++++++------
 1 file changed, 26 insertions(+), 6 deletions(-)

diff --git a/lvgl/src/core/lv_refr.c b/lvgl/src/core/lv_refr.c
index 300132f5f..ac5bb9508 100644
--- a/lvgl/src/core/lv_refr.c
+++ b/lvgl/src/core/lv_refr.c
@@ -391,9 +391,14 @@ void _lv_disp_refr_timer(lv_timer_t * tmr)
 #endif
 
 #if LV_USE_PERF_MONITOR && LV_USE_LABEL
+    static lv_style_t style;
+    lv_style_init(&style);
+    lv_style_set_text_font(&style, &lv_font_montserrat_24);
+
     lv_obj_t * perf_label = perf_monitor.perf_label;
     if(perf_label == NULL) {
         perf_label = lv_label_create(lv_layer_sys());
+        lv_obj_add_style(perf_label, &style, 0);
         lv_obj_set_style_bg_opa(perf_label, LV_OPA_50, 0);
         lv_obj_set_style_bg_color(perf_label, lv_color_black(), 0);
         lv_obj_set_style_text_color(perf_label, lv_color_white(), 0);
@@ -403,7 +408,7 @@ void _lv_disp_refr_timer(lv_timer_t * tmr)
         lv_obj_set_style_pad_right(perf_label, 3, 0);
         lv_obj_set_style_text_align(perf_label, LV_TEXT_ALIGN_RIGHT, 0);
         lv_label_set_text(perf_label, "?");
-        lv_obj_align(perf_label, LV_USE_PERF_MONITOR_POS, 0, 0);
+        lv_obj_align(perf_label, LV_ALIGN_OUT_LEFT_TOP, 0, 0);
         perf_monitor.perf_label = perf_label;
     }
 
@@ -449,9 +454,13 @@ void _lv_disp_refr_timer(lv_timer_t * tmr)
 
 #if LV_USE_MEM_MONITOR && LV_MEM_CUSTOM == 0 && LV_USE_LABEL
     lv_obj_t * mem_label = mem_monitor.mem_label;
-    static char kernelsize[256] = {0};
+    static char kernelcode[256] = {0};
+    static char rwdata[256] = {0};
+    static char rodata[256] = {0};
+    static char bss[256] = {0};
     if(mem_label == NULL) {
         mem_label = lv_label_create(lv_layer_sys());
+        lv_obj_add_style(mem_label, &style, 0);
         lv_obj_set_style_bg_opa(mem_label, LV_OPA_50, 0);
         lv_obj_set_style_bg_color(mem_label, lv_color_black(), 0);
         lv_obj_set_style_text_color(mem_label, lv_color_white(), 0);
@@ -463,8 +472,19 @@ void _lv_disp_refr_timer(lv_timer_t * tmr)
         lv_obj_align(mem_label, LV_USE_MEM_MONITOR_POS, 0, 0);
         mem_monitor.mem_label = mem_label;
 
-        FILE *fp = popen("dmesg | grep -Eo '[0-9]+K kernel code, [0-9]+K rwdata, [0-9]+K rodata' | tr '\n' ' '", "r");
-        fgets(kernelsize, sizeof(kernelsize), fp);
+        FILE *fp;
+        fp = popen("dmesg | grep -Eo '[0-9]+K kernel code' | tr '\n' ' '", "r");
+        fgets(kernelcode, sizeof(kernelcode), fp);
+        pclose(fp);
+        fp = popen("dmesg | grep -Eo '[0-9]+K rwdata' | tr '\n' ' '", "r");
+        fgets(rwdata, sizeof(rwdata), fp);
+        pclose(fp);
+        fp = popen("dmesg | grep -Eo '[0-9]+K rodata' | tr '\n' ' '", "r");
+        fgets(rodata, sizeof(rodata), fp);
+        pclose(fp);
+        fp = popen("dmesg | grep -Eo '[0-9]+K bss' | tr '\n' ' '", "r");
+        fgets(bss, sizeof(bss), fp);
+        pclose(fp);
     }
 
     if(lv_tick_elaps(mem_monitor.mem_last_time) > 300) {
@@ -475,9 +495,9 @@ void _lv_disp_refr_timer(lv_timer_t * tmr)
         long free_memory = info.freeram * info.mem_unit;
         long used_memory = total_memory - free_memory;
         lv_label_set_text_fmt(mem_label,
-                              "%lu MB used\n%s",
+                              "%lu MB used\n%s, %s\n%s, %s",
                               used_memory / 1024 / 1024,
-                              kernelsize);
+                              kernelcode, rwdata, rodata, bss);
     }
 #endif
 
-- 
2.45.2

