From 05f8f840d4daf56007e845b6c0d312d142548e02 Mon Sep 17 00:00:00 2001
From: U2FsdGVkX1 <U2FsdGVkX1@gmail.com>
Date: Tue, 30 Jul 2024 10:37:01 +0800
Subject: [PATCH] add abi version label

---
 src/core/lv_refr.c | 30 +++++++++++++++++++++++++++++-
 1 file changed, 29 insertions(+), 1 deletion(-)

diff --git a/lvgl/src/core/lv_refr.c b/lvgl/src/core/lv_refr.c
index 814315519..28e88a788 100644
--- a/lvgl/src/core/lv_refr.c
+++ b/lvgl/src/core/lv_refr.c
@@ -8,6 +8,8 @@
  *********************/
 #include <stddef.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include <sys/time.h>
 #include <sys/sysinfo.h>
 #include "lv_refr.h"
@@ -406,10 +408,36 @@ void _lv_disp_refr_timer(lv_timer_t * tmr)
         lv_obj_set_style_pad_bottom(perf_label, 3, 0);
         lv_obj_set_style_pad_left(perf_label, 3, 0);
         lv_obj_set_style_pad_right(perf_label, 3, 0);
-        lv_obj_set_style_text_align(perf_label, LV_TEXT_ALIGN_RIGHT, 0);
         lv_label_set_text(perf_label, "?");
         lv_obj_align(perf_label, LV_USE_PERF_MONITOR_POS, 0, 0);
         perf_monitor.perf_label = perf_label;
+
+        lv_obj_t * version_label = lv_label_create(lv_layer_sys());
+        lv_obj_add_style(version_label, &style, 0);
+        lv_obj_set_style_bg_opa(version_label, LV_OPA_50, 0);
+        lv_obj_set_style_bg_color(version_label, lv_color_black(), 0);
+        lv_obj_set_style_text_color(version_label, lv_color_white(), 0);
+        lv_obj_set_style_pad_top(version_label, 3, 0);
+        lv_obj_set_style_pad_bottom(version_label, 3, 0);
+        lv_obj_set_style_pad_left(version_label, 3, 0);
+        lv_obj_set_style_pad_right(version_label, 3, 0);
+        lv_obj_set_style_text_align(version_label, LV_TEXT_ALIGN_RIGHT, 0);
+        lv_obj_align(version_label, LV_ALIGN_TOP_RIGHT, 0, 0);
+
+        static char abiversion[256] = {0};
+        if(!system("zcat /proc/config.gz | grep -q CONFIG_ARCH_RV64ILP32=y")) {
+            strcat(abiversion, "kernel: s64ilp32\n");
+        } else if (!system("zcat /proc/config.gz | grep -q CONFIG_ARCH_RV64I=y")) {
+            strcat(abiversion, "kernel: s64lp64\n");
+        } else {
+            strcat(abiversion, "kernel: s32ilp32\n");
+        }
+        if(__SIZEOF_POINTER__ == 4) {
+            strcat(abiversion, "rootfs: u32ilp32");
+        } else {
+            strcat(abiversion, "rootfs: u64lp64");
+        }
+        lv_label_set_text(version_label, abiversion);
     }
 
     if(lv_tick_elaps(perf_monitor.perf_last_time) < 300) {
-- 
2.45.2

